Refresh.GameServer/Endpoints/Api/AuthenticationApiEndpoints.cs

```diff
@@ -0,0 +1,138 @@
+using System.Net;
+using System.Text.RegularExpressions;
+using Bunkum.CustomHttpListener.Parsing;
+using Bunkum.HttpServer;
+using Bunkum.HttpServer.Endpoints;
+using Bunkum.HttpServer.Responses;
+using Refresh.GameServer.Authentication;
+using Refresh.GameServer.Database;
+using Refresh.GameServer.Types.UserData;
+
+namespace Refresh.GameServer.Endpoints.Api;
+
+public partial class AuthenticationApiEndpoints : EndpointGroup
+{
+    // How many rounds to do for password hashing (BCrypt)
+    // 14 is ~1 second for logins and reset, which is fair because logins are a one-time thing
+    // 200 OK on POST '/api/v2/resetPassword' (1058ms)
+    // 200 OK on POST '/api/v2/auth' (1087ms)
+    //
+    // If increased, passwords will automatically be rehashed at login time to use the new WorkFactor
+    // If decreased, passwords will stay at higher WorkFactor until reset
+    private const int WorkFactor = 14;
+
+    [GeneratedRegex("^[a-f0-9]{128}$")]
+    private static partial Regex Sha512Regex();
+
+    [ApiEndpoint("auth", Method.Post)]
+    [Authentication(false)]
+    public Response Authenticate(RequestContext context, RealmDatabaseContext database, ApiAuthenticationRequest body)
+    {
+        GameUser? user = database.GetUserByUsername(body.Username);
+        if (user == null)
+        {
+            return new Response(new ApiErrorResponse("The username or password was incorrect."), ContentType.Json, HttpStatusCode.Forbidden);
+        }
+        
+        // database.SetUserPassword(user, null);
```

Comment:
leaving undocumented commented code here is probably a bad idea

Reply:
This was debugging code to quickly reset a user to a legacy user state to test password resets

I've removed it :+1: